src/libgap-api.h

```diff
@@ -1,24 +1,113 @@
-// LibGAP API - API for using GAP as shared library.
+//// LibGAP API - API for using GAP as shared library.
 
 #ifndef LIBGAP_API_H
 #define LIBGAP_API_H
 
-#include "gap.h"
+#include "system.h"
 
-typedef void (*CallbackFunc)(void);
 
-// Initialisation and finalization
+////
+//// Setup and initialisation
+////
 
-void GAP_Initialize(int          argc,
-                    char **      argv,
-                    char **      env,
-                    CallbackFunc markBagsCallback,
-                    CallbackFunc errorCallback);
+typedef void (*GAP_CallbackFunc)(void);
+
+extern void GAP_Initialize(int              argc,
+                           char **          argv,
+                           char **          env,
+                           GAP_CallbackFunc markBagsCallback,
+                           GAP_CallbackFunc errorCallback);
+
+
+////
+//// program evaluation and execution
+////
+
+// Evaluate a string of GAP commands
+//
+// To see an example of how to use this function see tst/testlibgap/basic.c
+extern Obj GAP_EvalString(const char * cmd);
+
+
+////
+//// variables
+////
+
+// Combines GVarName and ValGVar. For a given string, it returns the value
+// of the gvar with name <name>, or NULL if the global variable is not
+// defined.
+extern Obj GAP_ValueGlobalVariable(const char * name);
+
+
+////
+//// arithmetic
+////
+
+extern int GAP_EQ(Obj a, Obj b);
+extern int GAP_LT(Obj a, Obj b);
+extern int GAP_IN(Obj a, Obj b);
+
+extern Obj GAP_SUM(Obj a, Obj b);
+extern Obj GAP_DIFF(Obj a, Obj b);
+extern Obj GAP_PROD(Obj a, Obj b);
+extern Obj GAP_QUO(Obj a, Obj b);
+extern Obj GAP_LQUO(Obj a, Obj b);
+extern Obj GAP_POW(Obj a, Obj b);
+extern Obj GAP_COMM(Obj a, Obj b);
+extern Obj GAP_MOD(Obj a, Obj b);
```

Comment:
TODO: the above should get documentation comments. Also, there is still stuff from ariths.h missing which might be of interest -- or not... But hey, it's a start.